examples/flight_replay/flight_replay.py

```diff
@@ -7,122 +7,180 @@
         shorter = [messages[i] for i in xrange(0, num_points, step)]
         messages = shorter
+    return messages
+
+print("Generating waypoints from tlog...")
+messages = position_messages_from_tlog(args.tlog)
+print "Generated %d waypoints from tlog" % len(messages)
+
+#Start SITL if no connection string specified
+if args.connect:
+    connection_string = args.connect
+    sitl = None
+else:
+    start_lat = messages[0].lat/1.0e7
+    start_lon = messages[0].lon/1.0e7
 
-    print "Generating %s waypoints from replay..." % len(messages)
-    cmds = vehicle.commands
-    cmds.clear()
-    for i in xrange(0, len(messages)):
-        pt = messages[i]
-        lat = pt['lat']
-        lon = pt['lon']
-        # To prevent accidents we don't trust the altitude in the original flight, instead
-        # we just put in a conservative cruising altitude.
-        altitude = 30.0 # pt['alt']
-        cmd = Command( 0,
-                       0,
-                       0,
-                       mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT,
-                       mavutil.mavlink.MAV_CMD_NAV_WAYPOINT,
-                       0, 0, 0, 0, 0, 0,
-                       lat, lon, altitude)
-        cmds.add(cmd)
-    #Upload clear message and command messages to vehicle.
-    cmds.upload()
+    (sitl, connection_string) = start_default_sitl(lat=start_lat,lon=start_lon)
+
+# Connect to the Vehicle
+print 'Connecting to vehicle on: %s' % connection_string
+vehicle = connect(connection_string, wait_ready=True)
 
 
 # Now download the vehicle waypoints
 cmds = vehicle.commands
 cmds.wait_ready()
-mission_id = int(args.mission_id)
-max_freq = 0.1
-json = download_messages(mission_id, max_freq)
-print "JSON downloaded..."
-replay_mission(json)
 
 
-#Here you could actually replay the mission by
-#taking off and changing to AUTO mode.
-
+cmds = vehicle.commands
+cmds.clear()
+for i in xrange(0, len(messages)):
+    pt = messages[i]
+    print "Point: %d %d\n" % (pt.lat, pt.lon,)
+    lat = pt.lat
+    lon = pt.lon
+    # To prevent accidents we don't trust the altitude in the original flight, instead
+    # we just put in a conservative cruising altitude.
+    altitude = 30.0
+    cmd = Command( 0,
+                   0,
+                   0,
+                   mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT,
+                   mavutil.mavlink.MAV_CMD_NAV_WAYPOINT,
+                   0, 0, 0, 0, 0, 0,
+                   lat/1.0e7, lon/1.0e7, altitude)
+    cmds.add(cmd)
+
+#Upload clear message and command messages to vehicle.
+print("Uploading %d waypoints to vehicle..." % len(messages))
+cmds.upload()
+
+# Set mode to STABILISE for arming and takeoff:
+vehicle.mode = VehicleMode("GUIDED")
+
+while (vehicle.mode.name != "GUIDED"):
+    time.sleep(0.1)
+
+print("Arming vehicle")
+vehicle.armed = True
+
+while not vehicle.armed:      
+    print "Waiting for arming..."
+    time.sleep(1)
+
+print("Taking off")
+vehicle.simple_takeoff(30.0)
```

Comment:
This isn't "good code" since the command is asyncrhonous and if another command comes in it will be "replaced". It probably works because the time taken before someone is usually ready to start sending other commands is quite long ... so your discretion if it needs to change.